crates/assembly/src/assembler.rs

```diff
@@ -697,22 +710,25 @@ impl Assembler {
 
         let wrapper_proc = self.linker.get_procedure_unsafe(gid);
         let proc = wrapper_proc.unwrap_ast().unwrap_procedure();
-        let proc_body_id = if num_locals > 0 {
-            // For procedures with locals, we need to update fmp register before and after the
-            // procedure body is executed. Specifically:
-            // - to allocate procedure locals we need to increment fmp by the number of locals
-            //   (rounded up to the word size), and
-            // - to deallocate procedure locals we need to decrement it by the same amount.
-            let locals_frame = Felt::from(num_locals.next_multiple_of(WORD_SIZE as u16));
-            let wrapper = BodyWrapper {
-                prologue: vec![Operation::Push(locals_frame), Operation::FmpUpdate],
-                epilogue: vec![Operation::Push(-locals_frame), Operation::FmpUpdate],
-            };
-            self.compile_body(proc.iter(), &mut proc_ctx, Some(wrapper), mast_forest_builder)?
+        let body_wrapper = if proc_ctx.is_program_entrypoint() {
```

Comment:
Instead of Option<BodyWrapper> would it make sense to use `else {BodyWrapper::default()}`?